src/index.js

```diff
@@ -104,41 +124,202 @@ class ReactPlayerLoader extends React.Component {
+        }
+
+        // If the playlistVideoId changed and this is a playlist request, we
+        // need to search through the playlist items to find the correct
+        // starting index.
+        if (requestType === 'playlist' && changes.playlistVideoId) {
+          const i = findPlaylistItemIndex(data);
+
+          if (i > -1) {
+            data.startingIndex = i;
+          }
+        }
+
+        this.player.catalog.load(data);
+      };
+    };
+
+    let catalogParams;
+
+    if (this.player.usingPlugin('catalog')) {
+
+      // There is a new catalog sequence request. This takes precedence over
+      // other catalog updates because it is a different call.
+      if (changes.catalogSequence && this.props.catalogSequence) {
+        this.player.catalog.getLazySequence(this.props.catalogSequence, catalogCallback('sequence'), this.props.adConfigId);
+        return;
+      }
+
+      if (changes.videoId && this.props.videoId) {
+        catalogParams = {
+          type: 'video',
+          id: this.props.videoId
+        };
+      } else if (changes.playlistId && this.props.playlistId) {
+        catalogParams = {
+          type: 'playlist',
+          id: this.props.playlistId
+        };
+      } else if (changes.catalogSearch && this.props.catalogSearch) {
+        catalogParams = {
+          type: 'search',
+          q: this.props.catalogSearch
+        };
+      }
+    }
+
+    if (catalogParams) {
+      if (this.props.adConfigId) {
+        catalogParams.adConfigId = this.props.adConfigId;
+      }
+
+      if (this.props.deliveryConfigId) {
+        catalogParams.deliveryConfigId = this.props.deliveryConfigId;
+      }
+
+      // We use the callback style here to make tests simpler in IE11 (no need
+      // for a Promise polyfill).
+      this.player.catalog.get(catalogParams, catalogCallback(catalogParams.type));
+    } else if (
+      changes.playlistVideoId &&
+      this.props.playlistVideoId &&
+      this.player.usingPlugin('playlist')
+    ) {
+      const i = findPlaylistItemIndex(this.player.playlist());
+
+      if (i > -1) {
+        this.player.playlist.currentItem(i);
+      }
+    }
+  }
+
+  /**
+   * Called just after the component has mounted.
+   */
+  componentDidMount() {
+    this.isMounted_ = true;
+    this.loadPlayer();
+  }
+
+  /**
+   * Called when the component props are updated.
+   *
+   * Some prop changes may trigger special behavior (see `propChangeHandlers`),
+   * but if ANY prop is changed that is NOT handled, the player will be
+   * reloaded entirely.
+   *
+   * @param  {Object} prevProps
+   *         The previous props state before change.
+   */
+  componentDidUpdate(prevProps) {
+
+    // Calculate the prop changes.
+    const changes = Object.keys(prevProps).reduce((acc, key) => {
+      const previous = prevProps[key];
+      const current = this.props[key];
+
+      if (current !== previous) {
+        acc[key] = true;
+      }
+
+      return acc;
+    }, {});
```

Comment:
should the accumulator just be an array here?

Reply:
It's an object because checking if a property changed is much nicer like this:
```js
if (changes.foo) {
```
than this:
```js
if (changes.indexOf('foo') !== -1) {
```